ocsp/responder/responder.go

```diff
@@ -182,7 +182,10 @@ func (rs Responder) sampledError(format string, a ...interface{}) {
 // strings of repeated '/' into a single '/', which will break the base64
 // encoding.
 func (rs Responder) ServeHTTP(response http.ResponseWriter, request *http.Request) {
-	ctx := request.Context()
+	// We specifically ignore request.Context() because we would prefer for clients
+	// to not be able to cancel our operations in arbitrary places. Instead we
+	// start a new context, and apply timeouts in our various RPCs.
+	ctx := context.Background()
```

Comment:
Maybe add a `TODO(go1.22?): Use context.Detach() here instead` like we have elsewhere? In theory, the default request.Context() has useful tracing information on it already that we'd like to preserve if possible.

Reply:
What's the tracing information on a default request.Context()? I thought we added all that ourselves.

Reply:
The stdlib builds the request context [here](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/net/http/server.go;l=3047;drc=a37672bb7bef3d0163dd18f9f6914e2f829f1148;bpv=0;bpt=1). It immediately wraps it with [BaseContext](https://pkg.go.dev/net/http#Server), a function which can be used to add various useful key-values to the context. It then adds some useful values itself, such as [here](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/net/http/server.go;l=3057;drc=a37672bb7bef3d0163dd18f9f6914e2f829f1148;bpv=0;bpt=1) and [here](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/net/http/server.go;l=1844;drc=180bcad33dcd3d59443fe8eda5ae7556b1b2945b). Preserving those would be nice.